src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep.java

```diff
@@ -0,0 +1,81 @@
+package com.cloudbees.jenkins.plugins.sshagent;
+
+import hudson.Extension;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import java.io.Serializable;
+import java.util.List;
+
+public class SSHAgentStep extends AbstractStepImpl implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The {@link com.cloudbees.plugins.credentials.common.StandardUsernameCredentials#getId()}s of the credentials
+     * to use.
+     */
+    private List<String> credentials;
+
+    /**
+     * If a credentials is missed, the SSH Agent is launched anyway.
+     */
+    private boolean ignoreMissing;
+
+    /**
+     * Default parameterized constructor.
+     *
+     * @param credentials
+     * @param ignoreMissing
+     */
+    @DataBoundConstructor
+    public SSHAgentStep(final List<String> credentials, final boolean ignoreMissing) {
```

Comment:
You should not have `ignoreMissing` here. It is covered by a `@DataBoundSetter`.